ci/run_tests/run_tests_windows.py

```diff
@@ -1,14 +1,24 @@
 import os
 import argparse
 import subprocess
-import netifaces as ni
+import scapy.arch.windows
 
 TCPREPLAY_PATH = "tcpreplay-4.4.1-win"
 PCAP_FILE_PATH = os.path.abspath(
     os.path.join("Tests", "Pcap++Test", "PcapExamples", "example.pcap")
 )
 
 
+def get_ip_by_guid(guid):
+    interfaces = scapy.arch.windows.get_windows_if_list()
+    for iface in interfaces:
+        if iface["guid"] == guid:
+            # Return the second IP address if it exists, otherwise return None
+            return iface["ips"][1] if len(iface["ips"]) > 1 else None
```

Comment:
Why do we always return the second address?
If we want to return the first IPv4 address, maybe we can do this instead:
```python
def validate_ipv4_address(address):
    try:
        IPv4Address(address)
        return True
    except ValueError:
        return False


def get_ip_by_guid(guid):
    interfaces = scapy.arch.windows.get_windows_if_list()
    for iface in interfaces:
        if iface["guid"] == guid:
            # Return the first IPv4 address if exists, otherwise return None
            return next(filter(validate_ipv4_address, iface["ips"]), None)

    # Return None if no matching interface is found
    return None
```

Reply:
@zhengfeihe  It will be better to leave some comments as well.

Reply:
@tigercosmos added